### ISSUE TITLE
`spawn ENAMETOOLONG` on Windows

### ISSUE THREAD

**Issue by chandlerkent** [karma-runner/grunt-karma#161]
I'm receiving the error `spawn ENAMETOOLONG` when running the karma task. I believe this is a Windows-specific issue.

I've debugged this a little and I think I've tracked it down, but am not sure on the proper fix (or if the fix would break other usages).

Basically, the list of files gets included twice in the arguments passed to grunt.util.spawn here:

https://github.com/karma-runner/grunt-karma/blob/master/tasks/grunt-karma.js#L121

Once because the files are included in the configuration specified by this.options():

https://github.com/karma-runner/grunt-karma/blob/master/tasks/grunt-karma.js#L29

And the second time because it is on this.data:

https://github.com/karma-runner/grunt-karma/blob/master/tasks/grunt-karma.js#L60

I added a line that looked like this that "fixes" it, but I don't know if this is safe in all cases:

```
if (data.options && data.options.files) {
  delete data.options.files;
}
```

I have ~240 files that are passed dynamically to the task configuration. Example:

```
grunt.config('karma.serve.options.files', scripts)
```

`scripts` above is dynamically generated by looking at the index.html file and including all the source files in order.

Any help is appreciated. Thanks!

**Comment by chandlerkent** [comment]
I added a `console.log(data)` line here:

https://github.com/karma-runner/grunt-karma/blob/master/tasks/grunt-karma.js#L114

And this is what `data` is on this line (note I removed many files for brevity, but I think you get the idea):

```json
{
	background: true,
	client: {
		args: ['--verbose'],
		useIframe: true,
		captureConsole: true
	},
	files: [
        '../src\\libraries\\jquery\\dist\\jquery.js',
        '../src\\libraries\\lodash\\lodash.js',
        '../src\\libraries\\moment\\min\\moment-with-langs.js',
        '../src\\libraries\\angular\\angular.js',
        '200+ other files omitted for brevity',
        'unit/**/*.js'
    ],
	configFile: 'c: \\Dev\\Git\\MarketingOps.UI\\test\\karma.conf.js',
	singleRun: false,
	logLevel: 'DEBUG',
	options: {
		files: [
            '../src\\libraries\\jquery\\dist\\jquery.js',
            '../src\\libraries\\lodash\\lodash.js',
            '../src\\libraries\\moment\\min\\moment-with-langs.js',
            '../src\\libraries\\angular\\angular.js',
            '200+ other files omitted for brevity',
            'unit/**/*.js'
        ]
	}
}
```

**Comment by NicBright** [comment]
Hi there,

I ran into the same issue. Your hints pointed me to the right file so I was able to quickly verify what you've written (so thanks for the detailed description with line pointers :-)).

The solution you proposed is only a temporary work around though, as the problem will be back as soon as you start to have more files (I have about 650 files so I did not even try it).

I've fixed it differently and already proposed a PR:
https://github.com/karma-runner/grunt-karma/pull/163

Regards,
Nicolas
